### PR TITLE
Post type check

### PR DESCRIPTION
**Pull request by swissspidy**
I was just in a really strange situation. After splitting up my `cmb_meta_boxes`filters for the different post types I'm using, the values wouldn't be saved anymore.

Scenario:

Post Type A:
Group field (id: social) with fields website, twitter

Post Type B:
Group field (id: social) with fields website, twitter, facebook

In my case, the facebook information won't save because CMB takes the field from post type A (with the same ID) for the comparison. Since this meta box lacks the facebook field, it is ignored.

Of course there's the check that a meta box was shown on the page. But it should also check if it's the meta box of this post type. This change does just that and it works properly.

### CODE REVIEW

**Comment by swissspidy** [comment]
Can anyone review this? @mattheu perhaps?

**Comment by mikeselander** [comment]
Hey @swissspidy - thanks for submitting this fix - I'll look at this over the weekend or early next week 100%. 

Thanks for the reminder and bump!

**Comment by mikeselander** [comment]
Hey @swissspidy - so that reviewing it in September thing completely failed, sorry about that!

2 things we'd need fixed if you're up for it still:
1. There's a merge conflict that needs to be addressed
2. Could we get brackets around the `if` statement?

Thanks for submitting this!

**Comment by swissspidy** [comment]
@mikeselander No worries. Added brackets now and made the comparison strict.

File: class.cmb-meta-box.php [hunk 1]
```diff
@@ -478,6 +478,11 @@ function save( $post_id = 0 ) {
 			return $post_id;
 		}
 
+		// Verify this meta box is for the right post type
+		if ( ! in_array( get_post_type( $post_id ), (array) $this->_meta_box['pages'] ), true ) {
```
Comment (mikeselander) [inline class.cmb-meta-box.php]: @swissspidy you added the strict parameter outside of the `in_array` wrap causing a 500.

Should be:
`if ( ! in_array( get_post_type( $post_id ), (array) $this->_meta_box['pages'] , true ) )`

Comment (swissspidy) [inline class.cmb-meta-box.php]: @mikeselander Whoops, guess I was in a hurry there. Fixed now.

**Pull request merged**